codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -270,11 +279,35 @@ private void writeFromStreamChannel0(Object msg, boolean flush, ChannelPromise p
         }
     }
 
-    private ChannelFuture createStreamChannel(ChannelHandlerContext ctx, int streamId,
-                                      ChannelHandler handler) {
-        EventLoopGroup group = streamGroup != null ? streamGroup : ctx.channel().eventLoop();
-        Http2StreamChannel channel = new Http2StreamChannel(streamId);
+    /**
+     * Notifies any child streams of the read completion.
+     */
+    @Override
+    public void channelReadComplete(ChannelHandlerContext ctx) {
+        for (int i = 0; i < channelsToFireChildReadComplete.size(); i++) {
+            Http2StreamChannel childChannel = channelsToFireChildReadComplete.get(i);
+            // Clear early in case fireChildReadComplete() causes it to need to be re-processed
+            childChannel.inStreamsToFireChildReadComplete = false;
+            childChannel.fireChildReadComplete();
+        }
+        channelsToFireChildReadComplete.clear();
+    }
+
+    ChannelFuture createStreamChannel(Channel parentChannel, EventLoopGroup group, ChannelHandler handler,
+                                              Map<ChannelOption<?>, Object> options,
+                                              Map<AttributeKey<?>, Object> attrs,
+                                              int streamId) {
+        final Http2StreamChannel channel = new Http2StreamChannel(parentChannel);
+        if (Http2CodecUtil.isStreamIdValid(streamId)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Unnecessary use of fully qualified name 'Http2CodecUtil.isStreamIdValid' due to existing static import 'io.netty.handler.codec.http2.Http2CodecUtil.isStreamIdValid' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUnnecessaryFullyQualifiedName)